core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -399,16 +401,35 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
                     }
                     DBIO.seq(inserts: _*)
                   } andThen {
-                  DBIO.successful(destWorkspaceContext.workspace)
+                  DBIO.successful((sourceWorkspaceContext, destWorkspaceContext))
                 }
               }
             }
           }
+        }.map { case (sourceWorkspaceContext, destWorkspaceContext) =>
+          //we will fire and forget this. a more involved, but robust, solution involves using the Google Storage Transfer APIs
+          //in most of our use cases, these files should copy quickly enough for there to be no noticeable delay to the user
+          //we also don't want to block returning a response on this call because it's already a slow endpoint
+          destWorkspaceRequest.copyFilesWithPrefix.foreach { prefix =>
+            copyBucketFiles(sourceWorkspaceContext, destWorkspaceContext, prefix)
+          }
+
+          destWorkspaceContext.workspace
         }
       }
     }
   }
 
+  private def validateFileCopyPrefix(copyFilesWithPrefix: String): Unit = {
+    if(copyFilesWithPrefix.equalsIgnoreCase("")) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, """You may not specify an empty string for `copyFilesWithPrefix`. Did you mean to specify "/" or leave the field out entirely?"""))
```

Comment:
i chose to consider empty string an invalid value for a couple of reasons

1) if a user were to populate the payload in swagger ui it would include the empty string by default. if they don't remove it, they will copy everything from the root
2) specifying a prefix of "/" seems like a more intuitive way to say "i want everything" than the empty string

Reply:
Nit: `if (copyFilesWithPrefix.isEmpty)`

Treating empty string as invalid sounds fine to me